From 551353482a8039121966183048ea822832111326 Mon Sep 17 00:00:00 2001
From: Julian Bouzas <julian.bouzas@collabora.com>
Date: Thu, 13 Nov 2025 11:09:41 -0500
Subject: [PATCH] monitor/alsa: Also include alsa.* device properties for rule
 matching

UCM alsa nodes don't seem to have the 'alsa.*' properties from the device
included, which make it harder to match those nodes with alsa rules. This
patch adds all the 'alsa.*' properties in the UCM node to solve this.
---
 src/scripts/monitors/alsa.lua | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/scripts/monitors/alsa.lua b/src/scripts/monitors/alsa.lua
index 7173c543..1557997b 100644
--- a/src/scripts/monitors/alsa.lua
+++ b/src/scripts/monitors/alsa.lua
@@ -297,9 +297,9 @@ function createNode(parent, id, obj_type, factory, properties)
     properties["node.description"] = desc:gsub("(:)", " ")
   end
 
-  -- add api.alsa.card.* properties for rule matching purposes
+  -- add api.alsa.card.* and alsa.* properties for rule matching purposes
   for k, v in pairs(dev_props) do
-    if k:find("^api%.alsa%.card%..*") then
+    if k:find("^api%.alsa%.card%..*") or k:find("^alsa%..*") then
       properties[k] = v
     end
   end
-- 
2.51.2

